px4_ros2_cpp/src/components/mode_executor.cpp

```diff
@@ -269,17 +269,23 @@ void ModeExecutorBase::rtl(const CompletedCallback & on_completed)
   scheduleMode(ModeBase::kModeIDRtl, on_completed);
 }
 
-void ModeExecutorBase::arm(const CompletedCallback & on_completed)
+void ModeExecutorBase::arm(const CompletedCallback & on_completed, const bool & run_preflight_checks)
 {
   if (_is_armed) {
     on_completed(Result::Success);
     return;
   }
 
-  const Result result = sendCommandSync(
+  Result result = sendCommandSync(
     px4_msgs::msg::VehicleCommand::VEHICLE_CMD_COMPONENT_ARM_DISARM,
     1.f);
 
+  if (!run_preflight_checks)
```

Comment:
```suggestion
  if (!run_preflight_checks && result != Result::Success)
```
You could also use this:
```
const float param2 = run_preflight_checks ? NAN : 21196.f;
const result = sendCommandSync(px4_msgs::msg::VehicleCommand::VEHICLE_CMD_COMPONENT_ARM_DISARM,
    1.f, param2);
```